mureq.py

```diff
@@ -187,6 +188,22 @@ def _debugstr(self):
             print("<%d bytes binary data>" % (len(self.body),), file=buf)
         return buf.getvalue()
 
+    def raise_for_status(self):
+        """Raises :class:`HTTPException`, if one occurred."""
+
+        http_error_msg = ''
+        if 400 <= self.status_code < 500:
+            http_error_msg = f'{self.status_code} - Client error for url: {self.url}'
+
+        elif 500 <= self.status_code < 600:
+            http_error_msg = f'{self.status_code} - Server error for url: {self.url}'
+
+        if http_error_msg:
+            raise HTTPException(http_error_msg)
+
+    def json(self):
```

Comment:
would it make sense to cache the parsed JSON value? could save time in case there is a very large payload

Reply:
It's an interesting suggestion, but since Requests makes the client handle this, and the main goal here is parity with Requests, I think we'll do the same.